src/components/common/TextWithModalDetails.tsx

```diff
@@ -73,54 +73,26 @@ const styles = (theme: CustomTheme) =>
   });
 
 interface TextWithModalDetailsProps extends WithStyles<typeof styles> {
-  modalText: string;
   modalContent: string;
   modalHeading: string;
+  onDismiss: () => void;
 }
 
 export function TextWithModalDetails(props: TextWithModalDetailsProps) {
-  const [open, setOpen] = useState(false);
-  const { classes, modalText, modalContent, modalHeading } = props;
-
-  const handleClickOpen = () => {
-    setOpen(true);
-  };
-
-  const handleClose = () => {
-    setOpen(false);
-  };
-
-  const button = (
-    <Button
-      className={classes.openButton}
-      aria-label={modalText}
-      onClick={handleClickOpen}
-      disableFocusRipple={true}
-      disableRipple={true}
-    >
-      {modalText}
-    </Button>
-  );
+  const { classes, modalContent, modalHeading, onDismiss } = props;
 
   return (
     <span>
-      {button}
-      <Dialog
-        fullWidth
-        maxWidth="md"
-        onClose={handleClose}
-        aria-labelledby="customized-dialog-title"
-        open={open}
-      >
+      <Dialog fullWidth maxWidth="md" onClose={onDismiss} open={true}>
         <DialogTitle className={classes.dialogTitle} id="customized-dialog-title">
           {modalHeading}
-          <IconButton aria-label="Close" className={classes.closeButton} onClick={handleClose}>
+          <IconButton aria-label="Close" className={classes.closeButton} onClick={onDismiss}>
```

Comment:
why is the onDismiss method in the props now instead of defined in the component? And why the name change from handleClose?

Reply:
In order to have the flow work for making the API call, I had to pull the button out. So now the open state is handled by the parent, rather than the component itself. (I could fold the API call into this component, but it would make it less generalizeable.) It may be worth renaming the component?